app/lib/receipt.rb

```diff
@@ -0,0 +1,48 @@
+require_relative 'menu'
+require_relative 'discount'
+
+
+class Receipt
+
+  attr_reader :tax, :order
+
+  include Menu
+  include Discounts
+
+  def initialize(order, tax = 0)
+    @tax = tax
+    @order = order
+    menu_read_price_list
+  end
+
+  def price_of(item)
+    price_list[item]
+  end
+
+  def line_price(order_line)
+    order_line[:quantity] * price_of(order_line[:item])
+  end
+
+  def net_total
+    order.inject(0) { |memo, line| memo + line_price(line) }
+  end
+
+  def tax_total
+    (tax / 100 * (net_total - discount_total)).round(2)
+  end
+
+  def total_due
+    net_total - discount_total + tax_total
+  end
+
+  def print_receipt
```

Comment:
You're mixing responsibilities here, I'd argue that the receipt should just be doing this part which is generating the output of the receipt - calculating the totals I think is something that the order should do - probably delegating to a Calculator object ideally